### PR TITLE
feat(redis): add customIncrTtlLuaScript option

### PR DESCRIPTION
**Pull request by roggervalf**
Allow providing a custom incrTtlLuaScript

### CODE REVIEW

**Comment by roggervalf** [comment]
pls @animir when you get some time

**Comment by animir** [comment]
@roggervalf Hi. As I see you would like to double TTL of a key whenever consumes more than `points`? Do you need it to avoid extra requests to Redis? I am asking as I would like to understand your use case.

**Comment by roggervalf** [comment]
hey @animir, yes but more than doble it it would take how many point batches arento calculate the next delay, because when you already have the first batch of points, those are good to have the regular delay but for the secondbatch they will reach the same rate limit as the first batch came first

**Comment by roggervalf** [comment]
so the delay for the next points depending on which batch they are included need to have extra delays

**Comment by roggervalf** [comment]
pls @animir let me know if you have questions about it

**Comment by animir** [comment]
@roggervalf I do have doubts about bringing this customization to the project. 
It would be hard to explain how it works.
And I believe nobody, except you, would use it.

Could you implement it a different way? I suggest to consider settings custom Lua script from outside?
If there would be an ability to set custom Lua script when create RateLimiterRedis you could use you custom code there.
Let me know what you think.

**Comment by roggervalf** [comment]
sure I can do it

File: lib/RateLimiterRedis.js [hunk 1]
```diff
@@ -105,7 +106,7 @@ class RateLimiterRedis extends RateLimiterStoreAbstract {
     if (secDuration > 0) {
       if(!this.useRedisPackage && !this.useRedis3AndLowerPackage){
         return this.client.rlflxIncr(
-          [rlKey].concat([String(points), String(secDuration)]));
```
Comment (roggervalf) [inline lib/RateLimiterRedis.js:108]: as I'm passing points as a new arg and not as a new key, we should be good

File: lib/RateLimiterRedis.js [hunk 2]
```diff
@@ -118,15 +119,15 @@ class RateLimiterRedis extends RateLimiterStoreAbstract {
           };
 
           if (typeof this.client.rlflxIncr === 'function') {
-            this.client.rlflxIncr(rlKey, points, secDuration, incrCallback);
+            this.client.rlflxIncr(rlKey, points, secDuration, this.points, incrCallback);
```
Comment (roggervalf) [inline lib/RateLimiterRedis.js:122]: same as above and in the other cases

**Comment by roggervalf** [comment]
it should be ready now, pls when you get some time @animir

File: lib/RateLimiterRedis.js (hunk 1 above)
Comment (animir) [reply lib/RateLimiterRedis.js:108]: it looks good to me.

**Review by animir — APPROVED** [review]

**Comment by animir** [comment]
@roggervalf Published in `5.0.1`.

**Pull request merged**